### PR TITLE
London Class 10 | Ahmed Adam | JavaScript-Core-2-Coursework-Week3

### PR DESCRIPTION
**Pull request by ahmedsaif2002**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-exercises/A-reading-list/script.js [hunk 1]
```diff
@@ -25,4 +21,36 @@ const books = [
...
+    ul.appendChild(li);
+
+    if (book.alreadyRead) {
+      li.classList.add("read");
+      li.style.backgroundColor = "green";
+    
+    } else { 
+      li.classList.add("unread");
+      li.style.backgroundColor = "red";
+    }
```
Comment (jarrodruby) [inline 1-exercises/A-reading-list/script.js]:
```javascript
li.classList.add(book.alreadyRead ? 'read' : 'unread'  )
```

File: 1-exercises/B-highlight-words/script.js [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+  words.forEach((word) => {
+    // create span element
+    const span = document.createElement("span");
+    span.innerText = word + " ";
+    span.addEventListener("click", () => {
+      if (select.value === "none") {
+        span.style.backgroundColor = "";
+      } else {
+        span.style.backgroundColor = select.value;
+      }
```
Comment (jarrodruby) [inline 1-exercises/B-highlight-words/script.js]:
```javascript
span.style.backgroundColor = select.value  === "none" ? "" : select.value
```

File: 2-projects/1-todo-list/script.js [hunk 3]
```diff
@@ -1,6 +1,41 @@
...
+
+    let completeBtn = document.createElement("i");
+    completeBtn.classList.add("fa", "fa-check");
+    completeBtn.setAttribute("aria-hidden", "true");
+    completeBtn.addEventListener("click", function () {
+      if (li.style.textDecoration === "line-through") {
+        li.style.textDecoration = "none";
+      } else {
+        li.style.textDecoration = "line-through";
+      }
```
Comment (jarrodruby) [inline 2-projects/1-todo-list/script.js]: Use the tenary here as well

File: 2-projects/1-todo-list/script.js [hunk 4]
```diff
@@ -13,13 +48,32 @@ let todos = [
...
+  function deleteAllCompletedTodos() {
+    const deleteButton = document.getElementById("remove-all-completed");
+    deleteButton.addEventListener("click", function(){
+      const list = document.getElementById("todo-list");
+      const todos = list.getElementsByTagName("li");
+      for(let i = 0; i < todos.length; i++){
+        if(todos[i].style.textDecoration === "line-through"){
+          todos[i].remove();
+        }
+      }
```
Comment (jarrodruby) [inline 2-projects/1-todo-list/script.js]: Can this be a `forEach`

File: 1-exercises/A-reading-list/style.css [hunk 5]
```diff
@@ -11,6 +11,27 @@ body {
...
+  margin-bottom: 20px;
+}
+#reading-list img {
+  width: 100px;
+  height: auto;
+  margin-left: 20px;
+}
+#reading-list p {
+  font-size: 18px;
+  margin: 0;
```
Comment (jarrodruby) [inline 1-exercises/A-reading-list/style.css]: try removing `margin: 0;` as it should inherit from the parent I think

**Review by jarrodruby — APPROVED** [review]

File: 1-exercises/A-reading-list/script.js (hunk 1 above)
Comment (ahmedsaif2002) [reply 1-exercises/A-reading-list/script.js]: Thank you for add,  ternary operator has been modified

File: 1-exercises/B-highlight-words/script.js (hunk 2 above)
Comment (ahmedsaif2002) [reply 1-exercises/B-highlight-words/script.js]: done as well.

File: 2-projects/1-todo-list/script.js (hunk 3 above)
Comment (ahmedsaif2002) [reply 2-projects/1-todo-list/script.js]: yes sure i will

File: 2-projects/1-todo-list/script.js (hunk 4 above)
Comment (ahmedsaif2002) [reply 2-projects/1-todo-list/script.js]: Done!

File: 1-exercises/A-reading-list/style.css (hunk 5 above)
Comment (ahmedsaif2002) [reply 1-exercises/A-reading-list/style.css]: margin its has been removed

File: 2-projects/1-todo-list/script.js [hunk 6]
```diff
@@ -1,6 +1,39 @@
...
+    let badge = document.createElement("span");
+    badge.classList.add("badge", "bg-primary", "rounded-pill");
+    li.appendChild(badge);
+
+    let completeBtn = document.createElement("i");
+    completeBtn.classList.add("fa", "fa-check");
+    completeBtn.setAttribute("aria-hidden", "true");
+    completeBtn.addEventListener("click", function () {
+     li.style.textDecoration =
+     li.style.textDecoration === "line-through" ? "none" : "line-through";
```
Comment (jarrodruby) [inline 2-projects/1-todo-list/script.js]: I think you are missing something here

**Review by jarrodruby — APPROVED** [review]

File: 2-projects/1-todo-list/script.js (hunk 6 above)
Comment (ahmedsaif2002) [reply 2-projects/1-todo-list/script.js]: Thank you for you feedback , its one line i used assignment operator and strict equality operator.
like this  `li.style.textDecoration = li.style.textDecoration === "line-through" ? "none" : "line-through";`

**Pull request closed without merging**